main/src/motor/FocusMotor.cpp

```diff
@@ -1,808 +1,723 @@
-	{
-		// frameclock
-
-		int stepCounterPixel = 0;
-		int stepCounterLine = 0;
-		bool directionX = 0;
-		int iPixel = 0;
 
-		// Set pins high simultaenously at frame start
-		uint32_t mask = (1 << pinConfig.DIGITAL_OUT_1) | (1 << pinConfig.DIGITAL_OUT_2) | (1 << pinConfig.DIGITAL_OUT_3);
-		GPIO.out_w1ts = mask; // all high
-		bool directionY = 0;
-		for (int iLine = 0; iLine < nStepsLine; iLine += dStepsLine)
+#ifdef WIFI
+	void sendUpdateToClients(void *p)
+	{
+		for (;;)
 		{
-			// for every line all high
-			if (iLine > 0)
+			cJSON *root = cJSON_CreateObject();
+			cJSON *stprs = cJSON_CreateArray();
+			cJSON_AddItemToObject(root, key_steppers, stprs);
+			int added = 0;
+			for (int i = 0; i < 4; i++)
 			{
-				mask = (1 << pinConfig.DIGITAL_OUT_1) | (1 << pinConfig.DIGITAL_OUT_2);
-				GPIO.out_w1ts = mask; // all high
-			}
-
-			for (iPixel = 0; iPixel < nStepsPixel; iPixel += dStepsPixel)
-			{
-
-				if (motor->stageScanningData->stopped)
+				if (!data[i]->stopped)
 				{
-					break;
+#ifdef USE_FASTACCEL
+					FAccelStep::updateData(i);
+#elif defined USE_ACCELSTEP
+					AccelStep::updateData(i);
+#endif
+					cJSON *item = cJSON_CreateObject();
+					cJSON_AddItemToArray(stprs, item);
+					cJSON_AddNumberToObject(item, key_stepperid, i);
+					cJSON_AddNumberToObject(item, key_position, data[i]->currentPosition);
+					cJSON_AddNumberToObject(item, "isDone", data[i]->stopped);
+					added++;
 				}
-
-				// pulses the pixel trigger (or switch off in the first round)
-				mask = (1 << pinConfig.DIGITAL_OUT_1);
-				GPIO.out_w1ts = mask; // all high
-
-				// Move X motor forward at even steps, backward at odd steps
-				// bool directionX = iLine % 2 == 0;
-				moveMotor(pinStpPixel, pinDirPixel, dStepsPixel, directionX, delayTimeStep);
-				// stepCounterPixel += (dStepsPixel * (directionX ? 1 : -1));
-
-				// all low
-				mask = (1 << pinConfig.DIGITAL_OUT_1) | (1 << pinConfig.DIGITAL_OUT_2) | (1 << pinConfig.DIGITAL_OUT_3);
-				GPIO.out_w1tc = mask; // all low
 			}
+			if (added > 0)
+			{
+#ifdef WIFI
+				WifiController::sendJsonWebSocketMsg(root);
+#endif
+				// print result - will that work in the case of an xTask?
+				Serial.println("++");
+				char *s = cJSON_Print(root);
+				Serial.println(s);
+				free(s);
+				Serial.println("--");
+			}
+			cJSON_Delete(root);
+			vTaskDelay(1000 / portTICK_PERIOD_MS);
+		}
+	}
+#endif
 
-			// move back x stepper by step counter
-			moveMotor(pinStpPixel, pinDirPixel, iPixel, !directionX, (2 + delayTimeStep) * 10);
-
-			// Move Y motor after each line
-			moveMotor(pinStpLine, pinDirLine, dStepsLine, directionY, delayTimeStep);
-			stepCounterLine += (dStepsLine * (directionY ? 1 : -1));
+	void startStepper(int i)
+	{
+		// @KillerInk should this become a build flag? pinConfig.IS_I2C_MASTER?
+		log_i("start stepper %i", i);
+		if (pinConfig.IS_I2C_MASTER)
+		{
+			sendMotorDataI2C(*data[i], i); // TODO: This cannot send two motor information simultaenosly
```

Comment:
it dont have to send two motor information at once

Reply:
Sorry, im Standing in the pipe ;)